src/modules/uORB/uORBDeviceMaster.cpp

```diff
@@ -119,22 +119,45 @@ uORB::DeviceMaster::advertise(const struct orb_metadata *meta, int *instance, in
 			delete node;
 
 			if (ret == -EEXIST) {
-				/* if the node exists already, get the existing one and check if
-				 * something has been published yet. */
+				/* if the node exists already, get the existing one and check if it's advertised. */
```

Comment:
I guess we don't expect that there is other error conditions other than `EEXIST`?

Reply:
Yes there could be. For example another out of memory.